contracts/strategies/DeltaStrategy.sol

```diff
@@ -50,122 +72,306 @@ contract DeltaStrategy is Ownable {
     // vault.startWithdrawPeriod
   }
 
+  ///////////////////
+  // VAULT ACTIONS //
+  ///////////////////
+
+  function setBoard(uint boardId) external onlyVault {
+    Board memory board = getBoard(boardId);
+    require(isValidBoard(board), "invalid board");
+    activeExpiry = board.expiry;
+  }
+
+  function returnFundsAndClearStrikes() external onlyVault {
+    ExchangeRateParams memory exchangeParams = getExchangeParams();
+    uint quoteBal = quoteAsset.balanceOf(address(this));
+    uint baseBal = baseAsset.balanceOf(address(this));
+
+    uint quoteReceived = 0;
+    if (_isBaseCollat()) {
+      // todo: double check this
+      uint minQuoteExpected = baseBal.multiplyDecimal(exchangeParams.spotPrice).multiplyDecimal(
+        DecimalMath.UNIT - exchangeParams.baseQuoteFeeRate
+      );
+      quoteReceived = exchangeFromExactBase(baseBal, minQuoteExpected);
+    }
+    require(quoteAsset.transfer(vault, quoteBal + quoteReceived), "failed to return funds from strategy");
+
+    _clearAllActiveStrikes();
+  }
+
   /**
-   * request trade detail according to the strategy.
+   * @dev convert size (denominated in standard sizes) to actual contract amount.
    */
-  function doTrade() external view returns (uint positionId, uint premium) {
-    // todo: check whether minInterval has passed
-    uint boardId = 1;
-    uint listingId = _getListing(boardId);
-    uint amount = _getTradeAmount();
-    uint minPremium = _getMinPremium(listingId);
+  function getRequiredCollateral(uint strikeId) external view returns (uint requiredCollat) {
+    Strike memory strike = getStrikes(_toDynamic(strikeId))[0];
+    uint sellAmount = currentStrategy.size;
+    ExchangeRateParams memory exchangeParams = getExchangeParams();
+
+    // sets the correct amount for this trade instance
+    // if existing collateral is at risk, most likely wont be trading the same strike
+    // can call reducePosition() if previous trades are below buffer
+    uint minBufferCollateral = _getBufferCollateral(
+      strike.strikePrice,
+      strike.expiry,
+      exchangeParams.spotPrice,
+      sellAmount
+    );
```

Comment:
feel like we should let this function also takes in the tokenId of the existing position, and just re-adjust the whole ratio to the target ratio, instead of requiring multiple step to bring the position back to normal. 

(This also help not wasting collateral when the min collateral for a strike has gone down, we might not need to add more collateral to get more yield.)

Reply:
Yeah I think you're right... Originally I kept this out to prioritize simplicity, but since you're the second person to suggest this - will make this change.

Reply:
Since it may take a while for vault to fully unwind all it's positions, we're going to make sure the whole position is always above bufferCollat, but not necessarily meeting targetCollat